python/ray/_private/worker.py

```diff
@@ -1745,20 +1758,6 @@ def shutdown(_exiting_interpreter: bool = False):
 
 atexit.register(shutdown, True)
 
-
-# TODO(edoakes): this should only be set in the driver.
-def sigterm_handler(signum, frame):
-    sys.exit(signum)
```

Comment:
Actually I'm wondering why do we need this custom sigterm handler? Isn't the default behavior just exiting the process. @rkooo567 do you know why?

Reply:
Edward told me that "presumably the registered atexit handlers don’t run without it"

Reply:
I don't really know (this is the legacy code). I think it is safe to keep it for now, but we can try experimentation.